File: BOJ/Graph/Silver/11724/11724.swift
func dfs(_ cur : Int) {
    visit[cur] = true
    for next in nodes[cur] {
        if visit[next] {continue}
        dfs(next)
    }
}
let input = readLine()!.split(separator: " ").map{Int(String($0))!}
let (n,m) = (input[0],input[1])
var nodes : [[Int]] = Array(repeating: [], count: n+1)
var visit : [Bool] = Array(repeating: false, count: n+1)
for _ in 0..<m {
    let node = readLine()!.split(separator: " ").map{Int(String($0))!}
    nodes[node[0]].append(node[1])
    nodes[node[1]].append(node[0])
}
var cnt = 0
for i in 1...n {
    if visit[i] {continue}
    cnt+=1
    dfs(i)
}
print(cnt)

File: BOJ/Graph/Silver/11724/11724.py
import sys
sys.setrecursionlimit(1000000)
def dfs(cur) :
    visit[cur] = True
    for next in nodes[cur] :
        if visit[next] : continue
        dfs(next)

n,m = map(int,input().split())
cnt = 0
nodes = [[] for _ in range(n+1)]
visit = [False for _ in range(n+1)]
for _ in range(m) :
    node = list(map(int,sys.stdin.readline().split()))
    nodes[node[0]].append(node[1])
    nodes[node[1]].append(node[0])
for i in range(1,n+1) :
    if visit[i] : continue
    cnt+=1
    dfs(i)
print(cnt)

